pkg/uroot/uroot.go

```diff
@@ -111,25 +111,21 @@ type Opts struct {
 	UseExistingInit bool
 }
 
-// ResolvePackagePaths takes a list of Go package import paths and directories
-// and turns them into exclusively import paths.
-//
-// Currently allowed formats:
-//   Go package imports; e.g. github.com/u-root/u-root/cmds/ls
-//   Paths to Go package directories; e.g. $GOPATH/src/github.com/u-root/u-root/cmds/ls
-//   Globs of paths to Go package directories; e.g. ./cmds/*
-func ResolvePackagePaths(env golang.Environ, pkgs []string) ([]string, error) {
-	var importPaths []string
+// Resolve package paths for one package import path string.
+func resolvePackagePath(env golang.Environ, pkg string) ([]string, error) {
+	// The empty string prefix is to search relative to the working directory
+	// TODO(subhav): Using env.SrcDirs includes the GOROOT. Dunno if that makes sense.
+	prefixes := append([]string{""}, env.SrcDirs()...)
 	// Resolve file system paths to package import paths.
-	for _, pkg := range pkgs {
-		matches, err := filepath.Glob(pkg)
+	for _, prefix := range prefixes {
+		path := filepath.Join(prefix, pkg)
+		matches, err := filepath.Glob(path)
 		if len(matches) == 0 || err != nil {
-			if _, perr := env.Package(pkg); perr != nil {
-				return nil, fmt.Errorf("%q is neither package or path/glob: %v / %v", pkg, err, perr)
-			}
-			importPaths = append(importPaths, pkg)
+			//TODO(subhav): Log the error?
```

Comment:
not necessary, since env.Package below will return an error and we'll know this happened.

Reply:
Removed.